src/Lecture2.hs

```diff
@@ -260,8 +321,22 @@ data EvalError
 {- | Having all this set up, we can finally implement an evaluation function.
 It returns either a successful evaluation result or an error.
 -}
+
+getResultInt :: Maybe Int -> Int
+getResultInt (Just n) = n
+
+sumEither :: Either EvalError Int -> Either EvalError Int -> Either EvalError Int
+sumEither (Right x) (Right y) = Right (x+y)
+sumEither (Left (VariableNotFound str)) _ = Left (VariableNotFound str)
+sumEither _ (Left (VariableNotFound str)) = Left (VariableNotFound str)
+
 eval :: Variables -> Expr -> Either EvalError Int
-eval = error "TODO"
+eval vars expr = case expr of
+  Lit x -> Right x
+  Var x -> 
+    let result = lookup x vars
+    in (if (result == Nothing) then Left (VariableNotFound x) else Right (getResultInt result))
```

Comment:
You're using partial function `getResultInt` here and you compare your value with `Nothing` first. It's correct, however, such approach is considered an antipattern. You can use `case-of` instead to write correct-by-construction function:

```suggestion
    in case result of
            Nothing -> Left (VariableNotFound x)
            Just n  -> Right n
```